NullPointer exception on receiving intent

Crash occurs

```
org.mozilla.mozstumbler.client.mapview.MapActivity.mapNetworkConnectionChanged(MapActivity.java:332)
```

https://github.com/mozilla/MozStumbler/blob/v0.40.1.0/android/src/main/java/org/mozilla/mozstumbler/client/mapview/MapActivity.java#L332

```
java.lang.RuntimeException: Error receiving broadcast Intent { act=android.net.conn.CONNECTIVITY_CHANGE flg=0x10000000 (has extras) } in org.mozilla.mozstumbler.client.mapview.MapActivity$5@4089a2e8
at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:722)
at android.os.Handler.handleCallback(Handler.java:587)
at android.os.Handler.dispatchMessage(Handler.java:92)
at android.os.Looper.loop(Looper.java:130)
at android.app.ActivityThread.main(ActivityThread.java:3687)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:507)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:625)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
at org.mozilla.mozstumbler.client.mapview.MapActivity.mapNetworkConnectionChanged(MapActivity.java:332)
at org.mozilla.mozstumbler.client.mapview.MapActivity.access$900(MapActivity.java:62)
at org.mozilla.mozstumbler.client.mapview.MapActivity$5.onReceive(MapActivity.java:357)
at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:709)
... 9 more
java.lang.NullPointerException
at org.mozilla.mozstumbler.client.mapview.MapActivity.mapNetworkConnectionChanged(MapActivity.java:332)
at org.mozilla.mozstumbler.client.mapview.MapActivity.access$900(MapActivity.java:62)
at org.mozilla.mozstumbler.client.mapview.MapActivity$5.onReceive(MapActivity.java:357)
at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:709)
at android.os.Handler.handleCallback(Handler.java:587)
at android.os.Handler.dispatchMessage(Handler.java:92)
at android.os.Looper.loop(Looper.java:130)
at android.app.ActivityThread.main(ActivityThread.java:3687)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:507)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:625)
at dalvik.system.NativeStart.main(Native Method)
```

--- a/android/src/main/java/org/mozilla/mozstumbler/client/mapview/MapActivity.java
+++ b/android/src/main/java/org/mozilla/mozstumbler/client/mapview/MapActivity.java
@@ -337,7 +337,6 @@ public final class MapActivity extends android.support.v4.app.Fragment
         // can return NULL.  Lesson learned - never trust android framework.
         // https://github.com/mozilla/MozStumbler/issues/981
         if (activity == null) {
-            showMapNotAvailableMessage(NoMapAvailableMessage.eNoMapDueToNoAccessibleStorage);
             return;
         }
 

--- a/android/src/main/java/org/mozilla/mozstumbler/client/mapview/MapActivity.java
+++ b/android/src/main/java/org/mozilla/mozstumbler/client/mapview/MapActivity.java
@@ -17,6 +17,7 @@ import android.net.ConnectivityManager;
 import android.net.NetworkInfo;
 import android.os.Build;
 import android.os.Bundle;
+import android.support.v4.app.FragmentActivity;
 import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.MenuItem;
@@ -329,7 +330,18 @@ public final class MapActivity extends android.support.v4.app.Fragment
     }
 
     private void mapNetworkConnectionChanged() {
-        if (getActivity().getFilesDir() == null) {
+
+        FragmentActivity activity = getActivity();
+
+        // Amazingly, getActivity() while you're in a fragment
+        // can return NULL.  Lesson learned - never trust android framework.
+        // https://github.com/mozilla/MozStumbler/issues/981
+        if (activity == null) {
+            showMapNotAvailableMessage(NoMapAvailableMessage.eNoMapDueToNoAccessibleStorage);
+            return;
+        }
+
+        if (activity.getFilesDir() == null) {
             // Not the ideal spot for this check perhaps, but there is no point in checking
             // the network when storage is not available.
             showMapNotAvailableMessage(NoMapAvailableMessage.eNoMapDueToNoAccessibleStorage);

